packages/block/src/header.ts

```diff
@@ -546,13 +577,73 @@ export class BlockHeader {
         throw new Error('uncle block has a parent that is too old or too young')
       }
     }
+
+    // check if the block used too much gas
+    if (this.gasUsed.gt(this.gasLimit)) {
+      throw new Error('Invalid block: too much gas used')
+    }
+
+    if (this._common.isActivatedEIP(1559)) {
+      if (!this.baseFeePerGas) {
+        throw new Error('EIP1559 block has no base fee field')
+      }
+      const isInitialEIP1559Block = this._common.getEIPActivationBlockNumber(1559)!.eq(this.number)
+      if (isInitialEIP1559Block) {
+        const initialBaseFee = new BN(this._common.param('gasConfig', 'initialBaseFee'))
+        if (!this.baseFeePerGas!.eq(initialBaseFee)) {
+          throw new Error('Initial EIP1559 block does not have initial base fee')
+        }
+      } else {
+        // check if the base fee is correct
+        const expectedBaseFee = parentHeader.calcNextBaseFee()
+
+        if (!this.baseFeePerGas!.eq(expectedBaseFee)) {
+          throw new Error('Invalid block: base fee not correct')
```

Comment:
should we add some code coverage for these lines?

Reply:
We generally have to lift up coverage significantly for the PR - yes. Will therefore mark as resolved here, we just should generally look carefully through the Codecov reports.